src/dmangle.d

```diff
@@ -88,6 +94,9 @@ private immutable char[TMAX] mangleChar =
     //              X   // variadic T t...)
     //              Y   // variadic T t,...)
     //              Z   // not variadic, end of parameters
+    //              %   // Type backward reference
+    //              ?   // symbol backward reference
+    //              $   // identifier backward reference
```

Comment:
I would advise against using symbols, as they may cause invalid assembly or may have special meaning in the assembler.

For instance, `%` could start a comment anywhere on the line, or `$` can be used as statement separators.  I'd be much more comfortable using letters here.

We have `z` at least to add a number of extra meanings - maybe `zT`, `zS` and `zI` respectively?  (I assume these back references are like substitutions in C++).